### DISCUSSION TITLE
344. Reverse String

### DISCUSSION THREAD

**Discussion by mah-shamim** [mah-shamim/leet-code-in-php#121] [Q&A] [answered]
**Topics:** `Two Pointers`, `String`

Write a function that reverses a string. The input string is given as an array of characters `s`.

You must do this by modifying the input array [in-place](https://en.wikipedia.org/wiki/In-place_algorithm) with `O(1)` extra memory.

**Example 1:**

- **Input:** s = ["h","e","l","l","o"]
- **Output:** ["o","l","l","e","h"]

**Example 2:**

- **Input:** s = ["H","a","n","n","a","h"]
- **Output:** ["h","a","n","n","a","H"]

**Constraints:**

- <code>21 <= s.length <= 10<sup>5</sup></code>
- `s[i]` is a [printable ascii character](https://en.wikipedia.org/wiki/ASCII#Printable_characters).

**Hint:**
1. The entire logic for reversing a string is based on using the opposite directional two-pointer approach!

**Comment by mah-shamim** [comment, marked as answer]
To solve this problem, we can follow these steps:

Let's implement this solution in PHP: **[344. Reverse String](https://github.com/mah-shamim/leet-code-in-php/tree/main/algorithms/000344-reverse-string/solution.php)**

```php
<?php
function reverseString(&$s) {
    $left = 0;
    $right = count($s) - 1;

    while ($left < $right) {
        // Swap the characters
        $temp = $s[$left];
        $s[$left] = $s[$right];
        $s[$right] = $temp;

        // Move the pointers
        $left++;
        $right--;
    }
}

// Example 1:
$s1 = ["h", "e", "l", "l", "o"];
reverseString($s1);
print_r($s1); // Output: ["o", "l", "l", "e", "h"]

// Example 2:
$s2 = ["H", "a", "n", "n", "a", "h"];
reverseString($s2);
print_r($s2); // Output: ["h", "a", "n", "n", "a", "H"]
?>
```

### Explanation:

1. **Initialization**:
   - Set two pointers: `left` starting from the beginning (`0`) and `right` from the end (`count($s) - 1`).

2. **Swapping**:
   - Swap the elements at `left` and `right` positions.
   - Increment the `left` pointer and decrement the `right` pointer after each swap.

3. **Loop**:
   - Continue the swapping until `left` is less than `right`.

4. **In-Place Modification**:
   - The input array `$s` is modified in place, meaning the original array is reversed without using any additional arrays.

This solution has a time complexity of `O(n)` and a space complexity of `O(1)`, as required.

**Reply by basharul-siddike** [reply]
Thanks to reverse a string using the two-pointer approach

**Reply by mah-shamim** [reply]
Thnaks
